EventFilter/DTRawToDigi/plugins/DTuROSRawToDigi.cc

```diff
@@ -0,0 +1,538 @@
+
+
+  calcCRC(dataWord & 0xFFFFFFFF0000FFFF, newCRC);
+  if ( newCRC != CRC ) {
+    if ( debug_ ) edm::LogWarning("dturos_unpacker") 
+      << "Calculated CRC " << std::hex << newCRC << " differs from CRC in trailer " << std::hex << CRC;
+    return;
+  }
+
+
+  if ( lines != evtLgth ) {
+    if ( debug_ ) edm::LogWarning("dturos_unpacker") 
+      << "Number of words read != event length " << std::dec << lines << " " << evtLgth;
+    return;
+  }
+
+
+  fwords.settrailer(dataWord);
+  fwords.setevtlgth(evtLgth);
+
+
+  //--> analyze event
+
+ 
+  std::vector<long>::iterator DTuROSiterator = DTuROSWordContainer.begin();
+  std::vector<long>::iterator DTuROSitend = DTuROSWordContainer.end();
+
+
+  for (; DTuROSiterator != DTuROSitend; ++DTuROSiterator) {
+
+    DTuROSROSData rwords;
+
+
+    dataWord  = (*DTuROSiterator); // Header AMC 1
+
+
+    int slot       = ( dataWord >> 56 ) & 0xF;      // positions 56 -> 59
+
+    
+    if ( (slot < 1) || (slot > 12) ) {
+      if ( debug_ ) edm::LogWarning("dturos_unpacker") << "Slot " << std::dec << slot << " out of range (1-12) in crate " << crate;
+      break;
+    }
+
+
+    rwords.setslot(slot);
+    rwords.setheader1(dataWord);
+
+
+    ++DTuROSiterator;
+    dataWord  = (*DTuROSiterator); // Header AMC 2
+
+
+    rwords.setheader2(dataWord);
+
+
+    for (int k=2; k<slot_size[slot]-1; ++k) {
+
+
+      ++DTuROSiterator;
+      dataWord  = (*DTuROSiterator);
+      int selector  = ( dataWord >> 60 ) & 0xF; // positions 60 -> 63
+      int selector2 = ( dataWord >> 28 ) & 0x1; // position  28
+
+
+      if ( selector == 4 ) { // OK word 
+
+	if ( rwords.getokword1() ){ rwords.setokword2(dataWord); }
+	else { rwords.setokword1(dataWord); }
+
+      }
+      else {
+
+	if ( selector == 2 ) { // TDC word
+
+	  int tdcTime    = ( dataWord >> 32 ) & 0x3FFF; // positions  32 -> 45
+	  int tdcChannel = ( dataWord >> 46 ) & 0x1F;   // positions  46 -> 50
+	  int tdcId      = ( dataWord >> 51 ) & 0x3;    // positions  51 -> 52
+	  int link       = ( dataWord >> 53 ) & 0x7F;   // positions  53 -> 59
+
+
+	  int dduId = theDDU(crate, slot, link);
+	  int rosId = theROS(crate, slot, link);
+	  int robId = theROB(crate, slot, link);
+
+
+	  DTROChainCoding channelIndex(dduId, rosId, robId, tdcId, tdcChannel);
+	  if (hitOrder.find(channelIndex.getCode()) == hitOrder.end()) hitOrder[channelIndex.getCode()] = 0;
+	  else hitOrder[channelIndex.getCode()]++;
+
+
+	  int wheelId, stationId, sectorId, slId,layerId, cellId;
+	  if ( ! mapping->readOutToGeometry(dduId, rosId, robId, tdcId, tdcChannel,
+					    wheelId, stationId, sectorId, slId, layerId, cellId)) {
+
+	    DTWireId detId = DTWireId(wheelId, stationId, sectorId, slId, layerId, cellId);
+	    int wire = detId.wire();
+
+	    DTDigi digi(wire, tdcTime, hitOrder[channelIndex.getCode()]);
+	    digis.insertDigi(detId.layerId(),digi);
+
+	  }
+
+	}
+        else if ( selector == 3 ) { // error word
```

Comment:
Please indent the same as previous line, to improve readability
(Even better if you could indent with two spaces (as you do everywhere else) the whole block instead  of one (as you only do in the first nestings of the if-block starting at line 286)